pyiron_atomistics/atomistics/structure/atoms.py

```diff
@@ -2566,17 +2566,63 @@ def set_initial_magnetic_moments(self, magmoms=None):
         Set array of initial magnetic moments.
 
         Args:
-            magmoms (numpy.ndarray/list): List of magneric moments
+            magmoms (numpy.ndarray/list/dict/float/None): List, dict or single
+                value of magneric moments
+
+
+        Example I:
+
+        >>> from pyiron_atomistics import Project
+        >>> structure = Project('.').create.structure.bulk('Ni', cubic=True)
+        >>> structure[-1] = 'Fe'
+        >>> v_dict = {'Fe': 1, 'Ni': 2}
+        >>> structure.set_initial_magnetic_moments(v_dict)
+        >>> structure.get_initial_magnetic_moments()
+        array([2, 2, 2, 1])
+
+
+        Example II:
+
+        >>> from pyiron_atomistics import Project
+        >>> structure = Project('.').create.structure.bulk('Ni', cubic=True)
+        >>> structure[-1] = 'Fe'
+        >>> structure.set_initial_magnetic_moments(1)
+        >>> print(structure.get_initial_magnetic_moments())
+
+        Output:
+
+        >>> array([1, 1, 1, 1])
+
+        If you want to make it non-magnetic, set `None`:
+
+        >>> structure.set_initial_magnetic_moments(None)
+
+        If a list or an array of length 3 (e.g. [1, 2, 3]) is set and the
```

Comment:
This is a little implicit for my liking? I'd prefer to force the user to set a flag non-collinear when they wish to specify and then their code also becomes more readable overall too.